### ISSUE TITLE
Poor performance, multiple lost packages and high response times (rtl8723be)

### ISSUE THREAD

**Issue by alfonsodg** [lwfinger/rtlwifi_new#371]
Hi, i notice that the performance of my rtl8723be is terrible.  I have read every article and posts in every forum about the problems related to this card (by example the ant_sel trick), but until now no solution to this problem.  Also i have done multiple tests: kernel regressions (4.16,4.15,4.14,etc.), recompile modules, etc. Any ideas?

----
```
uname -a
Linux aldebaran 4.17.0-2-MANJARO #1 SMP PREEMPT Fri Jun 8 07:13:17 UTC 2018 x86_64 GNU/Linux
```
------
 ```
systool -vm rtl8723be
Module = "rtl8723be"

  Attributes:
    coresize            = "143360" 
    initsize            = "0"
    initstate           = "live"
    refcnt              = "0"
    srcversion          = "41E3D27D506CBEDC478A504"
    taint               = "O"
    uevent              = <store method only>

  Parameters:
    ant_sel             = "0"
    aspm                = "1"
    debug_level         = "0"
    debug_mask          = "0"
    disable_watchdog    = "Y"
    fwlps               = "N"
    ips                 = "N"
    msi                 = "N"
    swenc               = "Y"
    swlps               = "N"
```

------

```
lspci -v | grep Wireless
02:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8723BE PCIe Wireless Network Adapter
	Subsystem: Hewlett-Packard Company RTL8723BE PCIe Wireless Network Adapter
```
-------
```
ping 1.1.1.1
64 bytes from 1.1.1.1: icmp_seq=357 ttl=57 time=55.7 ms
64 bytes from 1.1.1.1: icmp_seq=358 ttl=57 time=180 ms
64 bytes from 1.1.1.1: icmp_seq=359 ttl=57 time=64.6 ms
64 bytes from 1.1.1.1: icmp_seq=360 ttl=57 time=4060 ms
64 bytes from 1.1.1.1: icmp_seq=361 ttl=57 time=3079 ms
64 bytes from 1.1.1.1: icmp_seq=362 ttl=57 time=2092 ms
64 bytes from 1.1.1.1: icmp_seq=363 ttl=57 time=1172 ms
64 bytes from 1.1.1.1: icmp_seq=364 ttl=57 time=192 ms
64 bytes from 1.1.1.1: icmp_seq=365 ttl=57 time=31.9 ms
64 bytes from 1.1.1.1: icmp_seq=366 ttl=57 time=83.7 ms
64 bytes from 1.1.1.1: icmp_seq=367 ttl=57 time=33.1 ms
64 bytes from 1.1.1.1: icmp_seq=368 ttl=57 time=47.5 ms
64 bytes from 1.1.1.1: icmp_seq=369 ttl=57 time=16.8 ms
64 bytes from 1.1.1.1: icmp_seq=370 ttl=57 time=46.3 ms
```

**Comment by lwfinger** [comment]
I do not want to read about ping times, lost packets, or any other symptom until you have posted signal strengths for the various possible values of ant_sel. That is not a trick! It is a programmatic way to overcome the problems introduced by cost-cutting manufacturers  that failed to correctly encode their EFUSE with the actual value of the connector to which their single antenna is attached.

**Comment by Krzmbrzl** [comment]
I am not sure if that really is related but you said you tried different kernel version until 4.14
Did you happen to try it with version 4.12 or below? Because I am also using the rtl8723be and I noticed that it almost quits working with any kernel version higher than 4.12 (see [my bug report](https://bugzilla.kernel.org/show_bug.cgi?id=200091) for further information on the symptoms).

Maybe you are experiencing the same or at least a similar issue...

**Comment by lwfinger** [comment]
I still do not see any signal strengths being posted. I'm closing this issue due to lack of response.

**Comment by dev-harsh1998** [comment]
@Krzmbrzl try toggling between the available antenna 1,2 and see if the issue still persists for you I exactly had the same issue but after switching the interface to antenna 1 the issue was over and the same scenario holds true for one of my friend @harry8242 use happens to be a rtl8723be user as well

**Comment by Krzmbrzl** [comment]
@dev-harsh1998 I believe I tried using both antennas without it having any effect... I will re-check that again though. Just to make sure...

**Comment by Krzmbrzl** [comment]
I am using a `.conf` in `/etc/modprobe.d/` in order to set the respective options and I unloaded the module via `modprobe -r` and re-loaded it afterwards. I believe I never did a restart in between though...
I will check if that makes any difference soon.

And while doing so I can look up the signal strength you've asked for. Here's the one of my current, working configuration (as listed by `iwconfig`):
`Link Quality=53/70  Signal level=-57 dBm`

**Comment by Krzmbrzl** [comment]
Okay I just checked with kernel 4.13 as it is currently found in the Kernel git and it didn't even find the connection to my wifi network (haven't moved my laptop so I'm still only 2m away from my router) with either antenna. Rebooting in between didn't change it either.

Coming back to 4.12 I get a signal strength of `Link Quality=24/70 Signal level=-86 dBm` with antenna 2 opposed to the previously posted `Link Quality=66/70 Signal level=-44 dBm` I got with antenna 1

**Comment by lwfinger** [comment]
The device should work with a signal of -47 dBm. It definitely would NOT work with -86 dBm. The latter is what you get with no antenna attached. You definitely have the "one antenna with an incorrectly programmed EFUSE" problem.

When the ant_sel module option was first created, a particular value of ant_sel might have worked. In your case with kernel 4.12, obviously that was a value of 1. The problem is that not only the driver code, but also the btcoex module is involved. The Realtek group that maintains btcoex made "improvements" that broke the driver stuff. This problem was fixed in kernel 4.17.0, and has been backported to 4.16.8+ and 4.14.40+. I find no indication that 4.13 has been fixed. It is not a version maintained in the "stable" category, thus fixes like this are not backported to it. That statement might not be true if you were using a distro-maintained kernel.

The conclusion is that you should use a 4.17.x, 4.16.x, or 4.14.x kernel and redo the signal strength measurement to determine the new, correct value for ant_sel. As noted above, a "modprobe -r/modprobe" sequence will probably not get a valid result. The NIC has a tendency to retain a value until the power is reset.

As I have stated earlier, I wish that I had never added the ant_sel option, and forced all of you to go back to HP for a proper fix, or forced you to open the computer and move the antenna cable to the other connector! Unfortunately, that is no longer an option. As Claire Boothe Luce stated, "No good deed goes unpunished!".

**Comment by Krzmbrzl** [comment]
@lwfinger just wanted to report back that the wifi-card is working as expected with the 4.17.6 kernel.
Signal strength: -40dBm while sitting at pretty much the same spot I measured the signal strength the last time.
